src/metadata/index.rs

```diff
@@ -0,0 +1,117 @@
+use elasticsearch::cat::CatIndicesParts;
+use elasticsearch::params::Time;
+use elasticsearch::{Elasticsearch, Error};
+use std::collections::HashMap;
+use tokio::sync::RwLock;
+
+use crate::{exporter_metrics::SUBSYSTEM_REQ_HISTOGRAM, Exporter};
+
+pub(crate) type IndexToMetadata = RwLock<IndexDataMap>;
+
+pub(crate) type IndexDataMap = HashMap<String, IndexMetadata>;
+
+/// Current Index metadata
+#[derive(Debug, Clone)]
+pub struct IndexMetadata {
+    pub last_hearbeat: chrono::DateTime<chrono::Utc>,
+}
+
+impl IndexMetadata {
+    pub fn new() -> Self {
+        Self {
+            last_hearbeat: chrono::Utc::now(),
+        }
+    }
+
+    /// Set heartbeat
+    pub fn reset_heartbeat(&mut self) -> &mut Self {
+        self.last_hearbeat = chrono::Utc::now();
+        self
+    }
+}
+
+/// Current Index metadata
+#[derive(Debug, Deserialize, Default)]
+pub(crate) struct CatIndexData {
+    /// index name
+    pub(crate) index: String,
+}
+
+pub(crate) async fn _build(client: &Elasticsearch) -> Result<IndexDataMap, Error> {
```

Comment:
why prefix method with an underscore? 🤔

Reply:
there were two methods _build and build :smile: remove underscore :+1: